src/server/pps/server/api_server.go

```diff
@@ -341,7 +341,22 @@ func (a *apiServer) CreateJob(ctx context.Context, request *ppsclient.CreateJobR
 		}
 		outputCommit, err = pfsAPIClient.StartCommit(ctx, startCommitRequest)
 		if err != nil {
-			return nil, err
+			if strings.Contains(err.Error(), "already exists") {
+				if _, err := pfsAPIClient.DeleteCommit(
+					ctx,
+					&pfsclient.DeleteCommitRequest{
+						Commit: client.NewCommit(parent.Repo.Name, strings.Split(parent.ID, "/")[0]),
```

Comment:
I think this is pretty unlikely to happen because the way `runPipeline` is implemented it's pretty hard for there to be 2 concurrent `CreateJob`s running for the same pipeline. Also because this is temporary and I don't want to spend too much more time on it.